### PR TITLE
Add experimental cpus config and use CircleCI env var

### LINKED ISSUE #6353: Build failing with `spawn ENOMEM` / `spawn E2BIG`
# Bug report

## Describe the bug

In some circumstances, `next build` is failing in CI environment (CircleCI) with the exception `Callback was already called'. 

There is a random factor and it's not happening 100% of the time, but more than 50% at the moment. 

This is triggered from the `neo-async` package which is used by `webpack`:

```
/home/circleci/project/node_modules/neo-async/async.js:14
    throw new Error('Callback was already called.');
    ^

Error: Callback was already called.
    at throwError (/home/circleci/project/node_modules/neo-async/async.js:14:11)
    at /home/circleci/project/node_modules/neo-async/async.js:2805:7
    at afterBuild (/home/circleci/project/node_modules/webpack/lib/Compilation.js:859:17)
    at processTicksAndRejections (internal/process/next_tick.js:74:9)
```

## Example

See for example: https://circleci.com/gh/opencollective/opencollective-frontend/5926

## Version

This is a new issue after upgrading to next 8.0.x, we never had it with next 7.0.x.

### PR DESCRIPTION
**Pull request by ijjk**
Also, added exposing `webpack` to the next.config.js `webpack` function

Fixes: #6353

### CODE REVIEW

File: packages/next-server/server/config.js [hunk 1]
```diff
@@ -21,7 +22,8 @@ const defaultConfig = {
     pagesBufferLength: 2
   },
   experimental: {
-    amp: false
+    amp: false,
+    cpus: (Number(process.env.CIRCLE_NODE_TOTAL) || os.cpus() || { length: 1 }).length
```
Comment (Timer) [inline packages/next-server/server/config.js]:
```suggestion
    cpus: Number(process.env.CIRCLE_NODE_TOTAL) || (os.cpus() || { length: 1 }).length
```

File: packages/next/build/webpack/plugins/terser-webpack-plugin/src/TaskRunner.js [hunk 2]
```diff
@@ -10,12 +10,12 @@ import minify from './minify';
...
 export default class TaskRunner {
-  constructor() {
+  constructor(cpus) {
     this.cacheDir = findCacheDir({ name: 'next-minifier' })
     // In some cases cpus() returns undefined
     // https://github.com/nodejs/node/issues/19022
-    const cpus = os.cpus() || { length: 1 };
-    this.maxConcurrentWorkers = cpus.length - 1
+    this.maxConcurrentWorkers = cpus
+    console.log('using maxWorkers', this.maxConcurrentWorkers);
```
Comment (Timer) [inline packages/next/build/webpack/plugins/terser-webpack-plugin/src/TaskRunner.js]:
```suggestion
```

**Review by Timer — CHANGES_REQUESTED** [review]

File: packages/next/build/webpack-config.ts [hunk 3]
```diff
@@ -298,7 +299,7 @@ export default function getBaseWebpackConfig (dir: string, {dev = false, isServe
   }
 
   if (typeof config.webpack === 'function') {
-    webpackConfig = config.webpack(webpackConfig, { dir, dev, isServer, buildId, config, defaultLoaders, totalPages })
+    webpackConfig = config.webpack(webpackConfig, { dir, dev, isServer, buildId, config, defaultLoaders, totalPages, webpack })
```
Comment (timneutkens) [inline packages/next/build/webpack-config.ts]: I think this is a great idea, but I'm not sure how this fits into the current PR 🤔

Comment (ijjk) [reply packages/next/build/webpack-config.ts]: I'm trying to fix #6353 and this is related to that

**Review by timneutkens — APPROVED** [review]

**Pull request merged**